pkg/bgpv1/manager/reconcile.go

```diff
@@ -492,6 +521,59 @@ endpointsLoop:
 	return ls
 }
 
+func (r *NeighborReconciler) fetchPeerPassword(sc *ServerWithConfig, n *v2alpha1api.CiliumBGPNeighbor) (string, error) {
+	l := log.WithFields(
+		logrus.Fields{
+			"component": "manager.fetchPeerPassword",
+		},
+	)
+	if n.AuthSecretRef != "" {
+		secret, err := r.fetchSecret(n.AuthSecretRef)
+		if err != nil {
+			return "", fmt.Errorf("failed to fetch secret %q: %w", n.AuthSecretRef, err)
```

Comment:
Shouldn't we handle the missing secret case here? The missing secret is an expected state, so we shouldn't treat it as an error here, right?

Reply:
Right, I've altered the logic now. It handles missing different to an error.